seaborn_objects_recipes/recipes/plotting.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+from matplotlib import style
+from dataclasses import dataclass
+import statsmodels.formula.api as smf
+import seaborn.objects as so  
+
+@dataclass
+class RegressionWithCI(so.PolyFit):
+    """
+Returns a seaborn.objects scatter plot with Dots and a linear
+regression with confidence intervals.
+
+:param alpha: Confidence inteval alpha.
+:param include_dots: Should data points be included in plot?
+"""
+    def __init__(self, order=1, gridsize=100, alpha=0.05, include_dots=True):
+        super().__init__(order=order, gridsize=gridsize)
+        self.alpha = alpha
+        self.include_dots = include_dots
+
+    def fit_and_predict(self, data, yvar, xvar):
+        if len(data) < 3:
+            raise ValueError("Insufficient data points to fit the model.")
+        # Filter out missing data and reset index to avoid issues with row indices
+        data = data.dropna(subset=[yvar, xvar]).reset_index(drop=True)
+        
+        # Fit the OLS model
+        model = smf.ols(f"{yvar} ~ {xvar}", data=data).fit()
```

Comment:
Ah, so here you aren't using the functions from `PolyFit`, which is why you aren't getting higher polynomials for free.